addons/account/models/account_reconcile_model.py

```diff
@@ -115,6 +115,76 @@ def _validate_amount(self):
                 except re.error:
                     raise UserError(_('The regex is not valid'))
 
+    def _prepare_aml_vals(self, partner):
+        self.ensure_one()
+
+        taxes = self.tax_ids
+        if taxes and partner:
+            fiscal_position = self.env['account.fiscal.position']._get_fiscal_position(partner)
+            if fiscal_position:
+                taxes = fiscal_position.map_tax(taxes)
+
+        return {
+            'name': self.label,
+            'account_id': self.account_id.id,
+            'partner_id': partner.id,
+            'analytic_account_id': self.analytic_account_id.id,
+            'analytic_tag_ids': [Command.set(self.analytic_tag_ids.ids)],
+            'tax_ids': [Command.set(taxes.ids)],
+            'reconcile_model_id': self.model_id.id,
+            'journal_id': self.journal_id.id,
+        }
+
+    def _apply_line_for_manual_widget(self, residual_amount_currency, partner, currency):
+        self.ensure_one()
+
+        if self.amount_type == 'percentage':
+            amount_currency = currency.round(residual_amount_currency * (self.amount / 100.0))
+        elif self.amount_type == 'fixed':
+            sign = 1 if residual_amount_currency > 0.0 else -1
+            amount_currency = currency.round(self.amount * sign)
```

Comment:
maybe add a comment explaining that other amount_types aren't relevant in this widget